server/utils/vectorDbProviders/lance/index.js

```diff
@@ -207,9 +207,9 @@ const LanceDb = {
 
           vectors.push(vectorRecord);
           submissions.push({
+            ...vectorRecord.metadata,
```

Comment:
spread first so that our values overwrite whatever might be colliding in metadata - namely `vector` and `id` key. This way what we define as a property will carry throughout the creation and upsert of the vectors.